libraries/AP_HAL_ChibiOS/hwdef/Aocoda-RC-H743Dual/hwdef.dat

```diff
@@ -0,0 +1,218 @@
+
+# board ID for firmware load
+APJ_BOARD_ID AP_HW_AOCODA-RC-H743DUAL
+
+# crystal frequency, setup to use external oscillator
+OSCILLATOR_HZ 8000000
+
+# flash size
+FLASH_SIZE_KB 2048
+
+env OPTIMIZE -Os
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 12
+define CH_CFG_ST_RESOLUTION 16
+
+# USB setup
+USB_STRING_MANUFACTURER "Aocoda-RC-H743Dual"
+
+# Now we define the pins that USB is connected on.
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# OTG USB Detect
+PE2 EXT_CS2 CS
+
+# These are the pins for SWD debugging with a STlinkv2 or black-magic probe.
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SPI1 for IMU1 (BMI270/MPU6000)
+PC15 IMU1_CS CS
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PD7 SPI1_MOSI SPI1
+
+# SPI2 for MAX7456 OSD
+PB12 MAX7456_CS CS
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+# SPI3 for SPI flash - W25N01GV
+PD3 FLASH_CS CS
+PB3 SPI3_SCK SPI3
+PB4 SPI3_MISO SPI3
+PB5 SPI3_MOSI SPI3
+
+# SPI4 for IMU2 (BMI270)
+PE11 IMU2_CS CS
+PE12 SPI4_SCK SPI4
+PE13 SPI4_MISO SPI4
+PE14 SPI4_MOSI SPI4
+
+# two I2C bus, order of I2C buses
+I2C_ORDER I2C2 I2C1
+
+# I2C1
+PB6 I2C1_SCL I2C1
+PB7 I2C1_SDA I2C1
+
+# I2C2
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# ADC for Power
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+define HAL_BATT_MONITOR_DEFAULT 4
+define HAL_BATT_VOLT_PIN 10
+define HAL_BATT_CURR_PIN 11
+define HAL_BATT_VOLT_SCALE 11.0
+define HAL_BATT_CURR_SCALE 40.0
+
+# ADC for airspeed
+PC4 PRESSURE_SENS ADC1 SCALE(2)
+define HAL_DEFAULT_AIRSPEED_PIN 4
+
+# ADC for rssi
+PC5 RSSI_ADC ADC1
+define BOARD_RSSI_ANA_PIN 8
+
+# Define LEDs
+# blue LED0 marked as ACT
+# green LED1 marked as B/E
+PE3 LED0 OUTPUT LOW GPIO(90) # blue
+PE4 LED1 OUTPUT LOW GPIO(91) # orange
+define HAL_GPIO_A_LED_PIN 91
+define HAL_GPIO_B_LED_PIN 90
+define HAL_GPIO_LED_OFF 1
+
+# The normal usage of this ordering is:
+# 1) SERIAL0: console (primary mavlink, usually USB)
+# 2) SERIAL1: RC input
+# 3) SERIAL2: GPS
+# 4) SERIAL3: VTX
+# 5) SERIAL4: Telemetry
+# 6) SERIAL5: not available
+# 7) SERIAL6: ESC Telemetry
+# 8) SERIAL7: spare
+# 9) SERIAL8: Optical Flow
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 EMPTY USART6 UART7 UART8
+
+# USART1 (RC input), SERIAL1
+PA9  USART1_TX USART1
+PA10 TIM1_CH3 TIM1 RCININT PULLDOWN LOW
+
+# as an alternative config setup the RX1 pin as a uart. This allows
+# for bi-directional UART based receiver protocols such as FPort
+# without any extra hardware
+PA10 USART1_RX USART1 ALT(1)
+
+# USART2 (GPS), SERIAL2
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+
+# USART3 (VTX), SERIAL3
+PD8 USART3_TX USART3
+PD9 USART3_RX USART3
+
+# UART4 (Telemetry), SERIAL4
+PB8 UART4_RX UART4
+PB9 UART4_TX UART4
+
+# USART6 (ESC Telemetry), SERIAL6
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+
+# UART7 (spare), SERIA7
+PE7 UART7_RX UART7
+PE8 UART7_TX UART7
+
+# UART8 (Optical Flow), SERIAL8

```

Comment:
```suggestion
# UART8 (spare), SERIAL8

```

Reply:
@Hwurzburg 

It seems some error with SerialProtocol_NONE. Any idea?

```
In file included from ../../libraries/AP_Math/AP_Math.h:9,
                 from ../../libraries/AP_SerialManager/AP_SerialManager.cpp:24:
./hwdef.h:65:34: error: 'SerialProtocol_NONE' was not declared in this scope; did you mean 'SerialProtocol_None'?
   65 | #define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_NONE
      |                                  ^~~~~~~~~~~~~~~~~~~
../../libraries/AP_Param/AP_Param.h:125:118: note: in definition of macro 'AP_GROUPINFO_FLAGS'
  125 | #define AP_GROUPINFO_FLAGS(name, idx, clazz, element, def, flags) { name, AP_VAROFFSET(clazz, element), {def_value : def}, flags, idx, AP_CLASSTYPE(clazz, element)}
      |                                                                                                                      ^~~
../../libraries/AP_SerialManager/AP_SerialManager.cpp:227:5: note: in expansion of macro 'AP_GROUPINFO'
  227 |     AP_GROUPINFO("4_PROTOCOL",  7, AP_SerialManager, state[4].protocol, DEFAULT_SERIAL4_PROTOCOL),
      |     ^~~~~~~~~~~~
../../libraries/AP_SerialManager/AP_SerialManager.cpp:227:73: note: in expansion of macro 'DEFAULT_SERIAL4_PROTOCOL'
  227 |     AP_GROUPINFO("4_PROTOCOL",  7, AP_SerialManager, state[4].protocol, DEFAULT_SERIAL4_PROTOCOL),
      |                                                                         ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.

Waf: Leaving directory `/home/daniel/ArduPilot/aocoda-ardupilot/build/Aocoda-RC-H743Dual'
Build failed
 -> task in 'objs/AP_SerialManager' failed (exit status 1):
        {task 139859653393456: cxx AP_SerialManager.cpp -> AP_SerialManager.cpp.0.o}
 (run with -v to display more information)
```

Reply:
My mistake, typo issue. Never mind.